dot/state/epoch.go

```diff
@@ -191,21 +191,25 @@ func (s *EpochState) GetEpochForBlock(header *types.Header) (uint64, error) {
 		return 0, errors.New("header is nil")
 	}
 
-	firstSlot, err := s.baseState.loadFirstSlot()
+	if header.Number == 1 {
+		return 0, nil
+	}
+
+	chainFirstSlotNumber, err := s.retrieveFirstNonOriginBlockSlot(header.Hash())
 	if err != nil {
-		return 0, err
+		return 0, fmt.Errorf("retrieving very first slot number: %w", err)
 	}
 
 	slotNumber, err := header.SlotNumber()
 	if err != nil {
 		return 0, fmt.Errorf("getting slot number: %w", err)
 	}
 
-	return (slotNumber - firstSlot) / s.epochLength, nil
+	return (slotNumber - chainFirstSlotNumber) / s.epochLength, nil
 }
 
-// SetEpochDataRaw sets the epoch data raw for a given epoch
-func (s *EpochState) SetEpochDataRaw(epoch uint64, raw *types.EpochDataRaw) error {
+// StoreEpochDataRaw sets the epoch data raw for a given epoch
+func (s *EpochState) StoreEpochDataRaw(epoch uint64, raw *types.EpochDataRaw) error {
```

Comment:
nite: Get/Set syntax is still fine in this case.  I don't think it really matters to differentiate whether or not it is persisted or not in this context.